sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -68,6 +68,30 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
         .get("mapreduce.output.fileoutputformat.compress.type"))
     }
 
+    fileSinkConf.tableInfo.getOutputFileFormatClassName match {
+      case formatName if formatName.toLowerCase.endsWith("parquetoutputformat") =>
+        val compressionConf = "parquet.compression"
+        val compressionCodec = getCompressionByPriority(
+          fileSinkConf,
+          compressionConf,
+          default = sparkSession.sessionState.conf.parquetCompressionCodec) match {
+          case "NONE" => "UNCOMPRESSED"
+          case _@x => x
+        }
+        hadoopConf.set(compressionConf, compressionCodec)
+      case formatName if formatName.endsWith("OrcOutputFormat") =>
+        val compressionConf = "orc.compress"
+        val compressionCodec = getCompressionByPriority(
+          fileSinkConf,
+          compressionConf,
+          default = sparkSession.sessionState.conf.orcCompressionCodec) match {
```

Comment:
I suggest to add a normalization logics for both ORC and Parquet. 

Check the `ParquetOptions`.`shortParquetCompressionCodecNames`